galsim/fft.py

```diff
@@ -0,0 +1,267 @@
+        - The size in each direction must be even. (Ideally 2^k or 3*2^k for speed, but this is
+          not required.)
+        - If it has a real dtype, it will be coerced to numpy.float64.
+        - If it hsa a complex dtype, it will be coerced to numpy.complex128.
+
+    The returned array will be complex with dtype numpy.complex128.
+
+    If shift_in is True, then this is equivalent to applying numpy.fft.fftshift to the input.
+
+        >>> ka1 = numpy.fft.fft2(numpy.fft.fftshift(a))
+        >>> ka2 = galsim.fft.fft2(a, shift_in=True)
+
+    If shift_out is True, then this is equivalent to applying numpy.fft.fftshift to the output.
+
+        >>> ka1 = numpy.fft.fftshift(numpy.fft.fft2(a))
+        >>> ka2 = galsim.fft.fft2(a, shift_out=True)
+
+    @param a            The input array to be transformed
+    @param shift_in     Whether to shift the input array so that the center is moved to (0,0).
+    @param shift_out    Whether to shift the output array so that the center is moved to (0,0).
+
+    @returns a complex numpy array
+    """
+    s = a.shape
+    if len(s) != 2:
+        raise ValueError("Input array must be 2D.")
+    M, N = s
+    Mo2 = M // 2
+    No2 = N // 2
+
+    if M != Mo2*2 or N != No2*2:
+        raise ValueError("Input array must have even sizes")
+
+    if a.dtype.kind == 'c':
+        a = a.astype(np.complex128, copy=False)
+        xim = galsim._galsim.ConstImageViewC(a, -No2, -Mo2)
+        kar = xim.cfft(shift_in=shift_in, shift_out=shift_out).array
+    else:
+        a = a.astype(np.float64, copy=False)
+        xim = galsim._galsim.ConstImageViewD(a, -No2, -Mo2)
+        if False:
+            # This works, but it's a bit slower.
+            kar = xim.cfft(shift_in=shift_in, shift_out=shift_out).array
+        else:
+            # Start with rfft2 version
+            rkar = xim.rfft(shift_in=shift_in, shift_out=shift_out).array
+            # This only returns kx >= 0.  Fill out the full image.
+            kar = np.empty( (M,N), dtype=np.complex128)
+            if shift_out:
+                kar[:,No2:N] = rkar[:,0:No2]
+                kar[0,0:No2] = rkar[0,No2:0:-1].conjugate()
+                kar[1:Mo2,0:No2] = rkar[M-1:Mo2:-1,No2:0:-1].conjugate()
+                kar[Mo2:M,0:No2] = rkar[Mo2:0:-1,No2:0:-1].conjugate()
+            else:
+                kar[:,0:No2] = rkar[:,0:No2]
+                kar[0,No2:N] = rkar[0,No2:0:-1].conjugate()
+                kar[1:M,No2:N] = rkar[M-1:0:-1,No2:0:-1].conjugate()
+    return kar
+
+
+def ifft2(a, shift_in=False, shift_out=False):
+    """Compute the 2-dimensional inverse discrete Fourier Transform.
+
+    For valid inputs, the result is equivalent to numpy.fft.ifft2(a), but usually faster.
+
+        >>> a1 = numpy.fft.ifft2(ka)
+        >>> a2 = galsim.fft.ifft2(ka)
+
+    Restrictions on this version vs the numpy version:
+
+        - The array must be 2-dimensional.
+        - It must be square.
```

Comment:
Thanks.  That was the case for my first pass at these functions, but then I enabled non-square inputs and forgot to update the doc.